src/main/java/com/minecolonies/coremod/commands/colonycommands/DoRaidNowCommand.java

```diff
@@ -45,31 +58,48 @@ public String getCommandUsage(@NotNull final ICommandSender sender)
     }
 
     @Override
-    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final ActionMenu actionMenu) throws CommandException
     {
-        if (sender instanceof EntityPlayer && !isPlayerOpped(sender))
+        final Colony colony = actionMenu.getColonyForArgument("colony");
+        if (colony == null)
         {
-            sender.sendMessage(new TextComponentString("Must be OP to use command"));
+            sender.sendMessage(NO_COLONY_FOUND_MESSAGE);
             return;
         }
+        executeShared(server, sender, colony);
+    }
 
+    @Override
+    public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
+    {
+        Colony colony = null;
         if (args.length != 0)
         {
-            final Colony colony = ColonyManager.getColony(Integer.parseInt(args[0]));
+            colony = ColonyManager.getColony(Integer.parseInt(args[0]));
             if (colony == null)
             {
-                sender.sendMessage(NO_COLONY_FOUND_MESSAGE_ID);
+                sender.sendMessage(NO_COLONY_FOUND_MESSAGE);
                 return;
             }
-
-            MobEventsUtils.barbarianEvent(colony.getWorld(), colony);
-
-            sender.sendMessage(SUCCESSFUL);
         }
         else
         {
             sender.sendMessage(NO_ARGUMENTS);
         }
+
+        executeShared(server, sender, colony);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') "NullPointerException" will be thrown when invoking method "executeShared()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2259)